src/Element.ts

```diff
@@ -1,250 +1,695 @@
+import { ParentNode, NonDocumentTypeChildNode, ChildNode } from './mixins';
+import { getChildren, getPreviousElementSibling, getNextElementSibling } from './mixins';
+import Attr from './Attr';
+import Document from './Document';
 import Node from './Node';
-
-import MutationRecord from './mutations/MutationRecord';
-import queueMutationRecord from './mutations/queueMutationRecord';
-
-export interface Attr {
-	name: string;
-	value: string;
-}
+import { getContext } from './context/Context';
+import { appendAttribute, changeAttribute, removeAttribute, replaceAttribute } from './util/attrMutations';
+import {
+	expectArity,
+	throwInUseAttributeError,
+	throwInvalidCharacterError,
+	throwNotFoundError
+} from './util/errorHelpers';
+import {
+	matchesNameProduction,
+	validateAndExtract,
+	locateNamespacePrefix,
+	XMLNS_NAMESPACE
+} from './util/namespaceHelpers';
+import { NodeType } from './util/NodeType';
+import { asNullableString, asObject } from './util/typeHelpers';
 
 /**
- * Internal helper used to check if the given node is an Element object.
- *
- * @param node Node to check
- *
- * @return Whether node is an element
+ * 3.9. Interface Element
  */
-function isElement (node?: Node | null): boolean {
-	return !!node && node.nodeType === Node.ELEMENT_NODE;
-}
+export default class Element extends Node implements ParentNode, NonDocumentTypeChildNode, ChildNode {
```

Comment:
Could you consider overriding Function.name as a static property? This should make a dumped object slightly more readable. Also, this will make the asObject  function simpler.